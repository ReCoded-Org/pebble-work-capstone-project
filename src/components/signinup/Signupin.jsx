import Image from "next/image";
import Link from "next/link";
import React, { useEffect, useState } from "react";
const Signupin = ({ title }) => {
    const [domLoaded, setDomLoaded] = useState(false);
    useEffect(() => {
        setDomLoaded(true);
    }, []);

    return (
        <>
            {domLoaded && (
                <div className='m-4 flex flex-col items-center justify-center gap-12  md:flex-row'>
                    <div className=' h-100 flex w-96 flex-col items-center justify-center text-center'>
                        <Image
                            src='/svg/signupinMan.svg'
                            alt='logo'
                            height={300}
                            width={300}
                        />
                    </div>
                    <div className='flex w-96 flex-col justify-center py-1 text-center'>
                        <h1 className='py-3 text-3xl font-semibold md:flex'>
                            {title}
                        </h1>
                        <div>
                            {title === "Sign up" && (
                                <div className=' flex flex-row justify-between py-1'>
                                    <div>
                                        <input
                                            style={{ width: "190px" }}
                                            className='focus:shadow-outline w-25 appearance-none rounded border py-2 px-3 leading-tight text-gray-700 shadow focus:outline-none'
                                            id='name'
                                            type='text'
                                            placeholder='Name'
                                        />
                                    </div>
                                    <div>
                                        <input
                                            style={{ width: "190px" }}
                                            className='focus:shadow-outline w-25 appearance-none rounded border py-2 px-3 leading-tight text-gray-700 shadow focus:outline-none'
                                            id='surname'
                                            type='text'
                                            placeholder='Surname'
                                        />
                                    </div>
                                </div>
                            )}
                            <div className='py-1'>
                                <input
                                    className='focus:shadow-outline w-96 appearance-none rounded border py-2 px-3 leading-tight text-gray-700 shadow focus:outline-none'
                                    id='email'
                                    type='text'
                                    placeholder='Email adress'
                                />
                            </div>
                            <div className='py-1'>
                                <input
                                    className='focus:shadow-outline w-96 appearance-none rounded border py-2 px-3 leading-tight text-gray-700 shadow focus:outline-none'
                                    id='password'
                                    type='password'
                                    placeholder='Password'
                                />
                            </div>
                            <p className='hidden text-xs italic text-red-500'>
                                Please choose a password.
                            </p>
                            {title === "Sign up" ? (
                                <p className='signin m-1 flex text-xs italic text-gray-700'>
                                    Already have an account?
                                    <Link href='/signin'>
                                        <div className='cursor-pointer border-b border-primary-200 pl-1 text-primary-200'>
                                            {title === "Sign up"
                                                ? "Sign in"
                                                : "/Sign up"}
                                        </div>
                                    </Link>
                                </p>
                            ) : (
                                <p className='signin m-1 flex text-xs italic text-gray-700'>
                                    You dont have an account?
                                    <Link href='/signup'>
                                        <div className='cursor-pointer border-b border-primary-200 pl-1 text-primary-200'>
                                            {title === "Sign up"
                                                ? "Sign in"
                                                : "Sign up"}
                                        </div>
                                    </Link>
                                </p>
                            )}
                            <div className=' py-1'>
                                <button className='w-80 rounded bg-primary-200 px-2 py-1 md:w-auto '>
                                    {title}
                                </button>
                            </div>
                            <div className='flex flex-row items-center justify-center py-1'>
                                <Image
                                    src='/svg/line.svg'
                                    alt='line'
                                    width={190}
                                    height={2}
                                />
                                <div className='text-sm text-gray-400'>OR</div>
                                <Image
                                    src='/svg/line.svg'
                                    alt='line'
                                    width={190}
                                    height={2}
                                />
                            </div>
                            <div className='flex justify-center py-1'>
                                <button className=' flex w-96 flex-row items-center justify-center rounded border-2 border-b-4  border-black bg-transparent py-1 font-normal text-black hover:border-primary-200 hover:text-primary-200'>
                                    <Image
                                        className='px-1'
                                        src='/svg/google.svg'
                                        alt='google'
                                        height={30}
                                        width={30}
                                    />
                                    Countinue with Google
                                </button>
                            </div>
                            <div className='flex justify-center py-1'>
                                <button className='flex w-96 flex-row items-center justify-center rounded border-2 border-b-4  border-black bg-transparent py-1 font-normal text-black hover:border-primary-200 hover:text-primary-200'>
                                    <Image
                                        style={{ paddingRight: "3px" }}
                                        src='/svg/twitter.svg'
                                        alt='twitter'
                                        height={28}
                                        width={28}
                                    />
                                    Countinue with Twitter
                                </button>
                            </div>
                        </div>
                    </div>
                </div>
            )}
        </>
    );
};

export default Signupin;
