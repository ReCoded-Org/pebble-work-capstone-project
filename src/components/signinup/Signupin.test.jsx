"use strict";

import renderer from "react-test-renderer";

import Signupin from "./Signupin";

it("renders correctly", () => {
    const tree = renderer.create(<Signupin title='Sign up' />).toJSON();
    expect(tree).toMatchSnapshot();
});
