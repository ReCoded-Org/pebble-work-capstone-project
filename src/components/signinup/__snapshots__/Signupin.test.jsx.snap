// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div
  className="m-4 flex flex-col items-center justify-center gap-12 md:flex-row"
>
  <div
    className=" h-100 flex w-96 flex-col items-center justify-center text-center"
  >
    <span
      style={
        Object {
          "background": "none",
          "border": 0,
          "boxSizing": "border-box",
          "display": "inline-block",
          "height": "initial",
          "margin": 0,
          "maxWidth": "100%",
          "opacity": 1,
          "overflow": "hidden",
          "padding": 0,
          "position": "relative",
          "width": "initial",
        }
      }
    >
      <span
        style={
          Object {
            "background": "none",
            "border": 0,
            "boxSizing": "border-box",
            "display": "block",
            "height": "initial",
            "margin": 0,
            "maxWidth": "100%",
            "opacity": 1,
            "padding": 0,
            "width": "initial",
          }
        }
      >
        <img
          alt=""
          aria-hidden={true}
          src="data:image/svg+xml,%3csvg%20xmlns=%27http://www.w3.org/2000/svg%27%20version=%271.1%27%20width=%27300%27%20height=%27300%27/%3e"
          style={
            Object {
              "background": "none",
              "border": 0,
              "display": "block",
              "height": "initial",
              "margin": 0,
              "maxWidth": "100%",
              "opacity": 1,
              "padding": 0,
              "width": "initial",
            }
          }
        />
      </span>
      <img
        alt="logo"
        data-nimg="intrinsic"
        decoding="async"
        onError={[Function]}
        onLoad={[Function]}
        src="data:image/gif;base64,R0lGODlhAQABAIAAAAAAAP///yH5BAEAAAAALAAAAAABAAEAAAIBRAA7"
        style={
          Object {
            "border": "none",
            "bottom": 0,
            "boxSizing": "border-box",
            "display": "block",
            "height": 0,
            "left": 0,
            "margin": "auto",
            "maxHeight": "100%",
            "maxWidth": "100%",
            "minHeight": "100%",
            "minWidth": "100%",
            "objectFit": undefined,
            "objectPosition": undefined,
            "padding": 0,
            "position": "absolute",
            "right": 0,
            "top": 0,
            "width": 0,
          }
        }
      />
      <noscript>
        <img
          alt="logo"
          data-nimg="intrinsic"
          decoding="async"
          loading="lazy"
          src="/svg/signupinMan.svg"
          srcSet="/svg/signupinMan.svg 1x, /svg/signupinMan.svg 2x"
          style={
            Object {
              "border": "none",
              "bottom": 0,
              "boxSizing": "border-box",
              "display": "block",
              "height": 0,
              "left": 0,
              "margin": "auto",
              "maxHeight": "100%",
              "maxWidth": "100%",
              "minHeight": "100%",
              "minWidth": "100%",
              "objectFit": undefined,
              "objectPosition": undefined,
              "padding": 0,
              "position": "absolute",
              "right": 0,
              "top": 0,
              "width": 0,
            }
          }
        />
      </noscript>
    </span>
  </div>
  <div
    className="flex w-96 flex-col justify-center py-1 text-center"
  >
    <h1
      className="py-3 text-3xl font-semibold md:flex"
    >
      Sign up
    </h1>
    <div>
      <div
        className=" flex flex-row justify-between py-1"
      >
        <div>
          <input
            className="focus:shadow-outline w-25 appearance-none rounded border py-2 px-3 leading-tight text-gray-700 shadow focus:outline-none"
            id="name"
            placeholder="Name"
            style={
              Object {
                "width": "190px",
              }
            }
            type="text"
          />
        </div>
        <div>
          <input
            className="focus:shadow-outline w-25 appearance-none rounded border py-2 px-3 leading-tight text-gray-700 shadow focus:outline-none"
            id="surname"
            placeholder="Surname"
            style={
              Object {
                "width": "190px",
              }
            }
            type="text"
          />
        </div>
      </div>
      <div
        className="py-1"
      >
        <input
          className="focus:shadow-outline w-96 appearance-none rounded border py-2 px-3 leading-tight text-gray-700 shadow focus:outline-none"
          id="email"
          placeholder="Email adress"
          type="text"
        />
      </div>
      <div
        className="py-1"
      >
        <input
          className="focus:shadow-outline w-96 appearance-none rounded border py-2 px-3 leading-tight text-gray-700 shadow focus:outline-none"
          id="password"
          placeholder="Password"
          type="password"
        />
      </div>
      <p
        className="hidden text-xs italic text-red-500"
      >
        Please choose a password.
      </p>
      <p
        className="signin m-1 flex text-xs italic text-gray-700"
      >
        Already have an account?
        <div
          className="cursor-pointer border-b border-primary-200 pl-1 text-primary-200"
          onClick={[Function]}
          onMouseEnter={[Function]}
        >
          Sign in
        </div>
      </p>
      <div
        className=" py-1"
      >
        <button
          className="w-80 rounded bg-primary-200 px-2 py-1 md:w-auto "
        >
          Sign up
        </button>
      </div>
      <div
        className="flex flex-row items-center justify-center py-1"
      >
        <span
          style={
            Object {
              "background": "none",
              "border": 0,
              "boxSizing": "border-box",
              "display": "inline-block",
              "height": "initial",
              "margin": 0,
              "maxWidth": "100%",
              "opacity": 1,
              "overflow": "hidden",
              "padding": 0,
              "position": "relative",
              "width": "initial",
            }
          }
        >
          <span
            style={
              Object {
                "background": "none",
                "border": 0,
                "boxSizing": "border-box",
                "display": "block",
                "height": "initial",
                "margin": 0,
                "maxWidth": "100%",
                "opacity": 1,
                "padding": 0,
                "width": "initial",
              }
            }
          >
            <img
              alt=""
              aria-hidden={true}
              src="data:image/svg+xml,%3csvg%20xmlns=%27http://www.w3.org/2000/svg%27%20version=%271.1%27%20width=%27190%27%20height=%272%27/%3e"
              style={
                Object {
                  "background": "none",
                  "border": 0,
                  "display": "block",
                  "height": "initial",
                  "margin": 0,
                  "maxWidth": "100%",
                  "opacity": 1,
                  "padding": 0,
                  "width": "initial",
                }
              }
            />
          </span>
          <img
            alt="line"
            data-nimg="intrinsic"
            decoding="async"
            onError={[Function]}
            onLoad={[Function]}
            src="data:image/gif;base64,R0lGODlhAQABAIAAAAAAAP///yH5BAEAAAAALAAAAAABAAEAAAIBRAA7"
            style={
              Object {
                "border": "none",
                "bottom": 0,
                "boxSizing": "border-box",
                "display": "block",
                "height": 0,
                "left": 0,
                "margin": "auto",
                "maxHeight": "100%",
                "maxWidth": "100%",
                "minHeight": "100%",
                "minWidth": "100%",
                "objectFit": undefined,
                "objectPosition": undefined,
                "padding": 0,
                "position": "absolute",
                "right": 0,
                "top": 0,
                "width": 0,
              }
            }
          />
          <noscript>
            <img
              alt="line"
              data-nimg="intrinsic"
              decoding="async"
              loading="lazy"
              src="/svg/line.svg"
              srcSet="/svg/line.svg 1x, /svg/line.svg 2x"
              style={
                Object {
                  "border": "none",
                  "bottom": 0,
                  "boxSizing": "border-box",
                  "display": "block",
                  "height": 0,
                  "left": 0,
                  "margin": "auto",
                  "maxHeight": "100%",
                  "maxWidth": "100%",
                  "minHeight": "100%",
                  "minWidth": "100%",
                  "objectFit": undefined,
                  "objectPosition": undefined,
                  "padding": 0,
                  "position": "absolute",
                  "right": 0,
                  "top": 0,
                  "width": 0,
                }
              }
            />
          </noscript>
        </span>
        <div
          className="text-sm text-gray-400"
        >
          OR
        </div>
        <span
          style={
            Object {
              "background": "none",
              "border": 0,
              "boxSizing": "border-box",
              "display": "inline-block",
              "height": "initial",
              "margin": 0,
              "maxWidth": "100%",
              "opacity": 1,
              "overflow": "hidden",
              "padding": 0,
              "position": "relative",
              "width": "initial",
            }
          }
        >
          <span
            style={
              Object {
                "background": "none",
                "border": 0,
                "boxSizing": "border-box",
                "display": "block",
                "height": "initial",
                "margin": 0,
                "maxWidth": "100%",
                "opacity": 1,
                "padding": 0,
                "width": "initial",
              }
            }
          >
            <img
              alt=""
              aria-hidden={true}
              src="data:image/svg+xml,%3csvg%20xmlns=%27http://www.w3.org/2000/svg%27%20version=%271.1%27%20width=%27190%27%20height=%272%27/%3e"
              style={
                Object {
                  "background": "none",
                  "border": 0,
                  "display": "block",
                  "height": "initial",
                  "margin": 0,
                  "maxWidth": "100%",
                  "opacity": 1,
                  "padding": 0,
                  "width": "initial",
                }
              }
            />
          </span>
          <img
            alt="line"
            data-nimg="intrinsic"
            decoding="async"
            onError={[Function]}
            onLoad={[Function]}
            src="data:image/gif;base64,R0lGODlhAQABAIAAAAAAAP///yH5BAEAAAAALAAAAAABAAEAAAIBRAA7"
            style={
              Object {
                "border": "none",
                "bottom": 0,
                "boxSizing": "border-box",
                "display": "block",
                "height": 0,
                "left": 0,
                "margin": "auto",
                "maxHeight": "100%",
                "maxWidth": "100%",
                "minHeight": "100%",
                "minWidth": "100%",
                "objectFit": undefined,
                "objectPosition": undefined,
                "padding": 0,
                "position": "absolute",
                "right": 0,
                "top": 0,
                "width": 0,
              }
            }
          />
          <noscript>
            <img
              alt="line"
              data-nimg="intrinsic"
              decoding="async"
              loading="lazy"
              src="/svg/line.svg"
              srcSet="/svg/line.svg 1x, /svg/line.svg 2x"
              style={
                Object {
                  "border": "none",
                  "bottom": 0,
                  "boxSizing": "border-box",
                  "display": "block",
                  "height": 0,
                  "left": 0,
                  "margin": "auto",
                  "maxHeight": "100%",
                  "maxWidth": "100%",
                  "minHeight": "100%",
                  "minWidth": "100%",
                  "objectFit": undefined,
                  "objectPosition": undefined,
                  "padding": 0,
                  "position": "absolute",
                  "right": 0,
                  "top": 0,
                  "width": 0,
                }
              }
            />
          </noscript>
        </span>
      </div>
      <div
        className="flex justify-center py-1"
      >
        <button
          className=" flex w-96 flex-row items-center justify-center rounded border-2 border-b-4 border-black bg-transparent py-1 font-normal text-black hover:border-primary-200 hover:text-primary-200"
        >
          <span
            style={
              Object {
                "background": "none",
                "border": 0,
                "boxSizing": "border-box",
                "display": "inline-block",
                "height": "initial",
                "margin": 0,
                "maxWidth": "100%",
                "opacity": 1,
                "overflow": "hidden",
                "padding": 0,
                "position": "relative",
                "width": "initial",
              }
            }
          >
            <span
              style={
                Object {
                  "background": "none",
                  "border": 0,
                  "boxSizing": "border-box",
                  "display": "block",
                  "height": "initial",
                  "margin": 0,
                  "maxWidth": "100%",
                  "opacity": 1,
                  "padding": 0,
                  "width": "initial",
                }
              }
            >
              <img
                alt=""
                aria-hidden={true}
                src="data:image/svg+xml,%3csvg%20xmlns=%27http://www.w3.org/2000/svg%27%20version=%271.1%27%20width=%2730%27%20height=%2730%27/%3e"
                style={
                  Object {
                    "background": "none",
                    "border": 0,
                    "display": "block",
                    "height": "initial",
                    "margin": 0,
                    "maxWidth": "100%",
                    "opacity": 1,
                    "padding": 0,
                    "width": "initial",
                  }
                }
              />
            </span>
            <img
              alt="google"
              className="px-1"
              data-nimg="intrinsic"
              decoding="async"
              onError={[Function]}
              onLoad={[Function]}
              src="data:image/gif;base64,R0lGODlhAQABAIAAAAAAAP///yH5BAEAAAAALAAAAAABAAEAAAIBRAA7"
              style={
                Object {
                  "border": "none",
                  "bottom": 0,
                  "boxSizing": "border-box",
                  "display": "block",
                  "height": 0,
                  "left": 0,
                  "margin": "auto",
                  "maxHeight": "100%",
                  "maxWidth": "100%",
                  "minHeight": "100%",
                  "minWidth": "100%",
                  "objectFit": undefined,
                  "objectPosition": undefined,
                  "padding": 0,
                  "position": "absolute",
                  "right": 0,
                  "top": 0,
                  "width": 0,
                }
              }
            />
            <noscript>
              <img
                alt="google"
                className="px-1"
                data-nimg="intrinsic"
                decoding="async"
                loading="lazy"
                src="/svg/google.svg"
                srcSet="/svg/google.svg 1x, /svg/google.svg 2x"
                style={
                  Object {
                    "border": "none",
                    "bottom": 0,
                    "boxSizing": "border-box",
                    "display": "block",
                    "height": 0,
                    "left": 0,
                    "margin": "auto",
                    "maxHeight": "100%",
                    "maxWidth": "100%",
                    "minHeight": "100%",
                    "minWidth": "100%",
                    "objectFit": undefined,
                    "objectPosition": undefined,
                    "padding": 0,
                    "position": "absolute",
                    "right": 0,
                    "top": 0,
                    "width": 0,
                  }
                }
              />
            </noscript>
          </span>
          Countinue with Google
        </button>
      </div>
      <div
        className="flex justify-center py-1"
      >
        <button
          className="flex w-96 flex-row items-center justify-center rounded border-2 border-b-4 border-black bg-transparent py-1 font-normal text-black hover:border-primary-200 hover:text-primary-200"
        >
          <span
            style={
              Object {
                "background": "none",
                "border": 0,
                "boxSizing": "border-box",
                "display": "inline-block",
                "height": "initial",
                "margin": 0,
                "maxWidth": "100%",
                "opacity": 1,
                "overflow": "hidden",
                "padding": 0,
                "position": "relative",
                "width": "initial",
              }
            }
          >
            <span
              style={
                Object {
                  "background": "none",
                  "border": 0,
                  "boxSizing": "border-box",
                  "display": "block",
                  "height": "initial",
                  "margin": 0,
                  "maxWidth": "100%",
                  "opacity": 1,
                  "padding": 0,
                  "width": "initial",
                }
              }
            >
              <img
                alt=""
                aria-hidden={true}
                src="data:image/svg+xml,%3csvg%20xmlns=%27http://www.w3.org/2000/svg%27%20version=%271.1%27%20width=%2728%27%20height=%2728%27/%3e"
                style={
                  Object {
                    "background": "none",
                    "border": 0,
                    "display": "block",
                    "height": "initial",
                    "margin": 0,
                    "maxWidth": "100%",
                    "opacity": 1,
                    "padding": 0,
                    "width": "initial",
                  }
                }
              />
            </span>
            <img
              alt="twitter"
              data-nimg="intrinsic"
              decoding="async"
              onError={[Function]}
              onLoad={[Function]}
              src="data:image/gif;base64,R0lGODlhAQABAIAAAAAAAP///yH5BAEAAAAALAAAAAABAAEAAAIBRAA7"
              style={
                Object {
                  "border": "none",
                  "bottom": 0,
                  "boxSizing": "border-box",
                  "display": "block",
                  "height": 0,
                  "left": 0,
                  "margin": "auto",
                  "maxHeight": "100%",
                  "maxWidth": "100%",
                  "minHeight": "100%",
                  "minWidth": "100%",
                  "objectFit": undefined,
                  "objectPosition": undefined,
                  "padding": 0,
                  "paddingRight": "3px",
                  "position": "absolute",
                  "right": 0,
                  "top": 0,
                  "width": 0,
                }
              }
            />
            <noscript>
              <img
                alt="twitter"
                data-nimg="intrinsic"
                decoding="async"
                loading="lazy"
                src="/svg/twitter.svg"
                srcSet="/svg/twitter.svg 1x, /svg/twitter.svg 2x"
                style={
                  Object {
                    "border": "none",
                    "bottom": 0,
                    "boxSizing": "border-box",
                    "display": "block",
                    "height": 0,
                    "left": 0,
                    "margin": "auto",
                    "maxHeight": "100%",
                    "maxWidth": "100%",
                    "minHeight": "100%",
                    "minWidth": "100%",
                    "objectFit": undefined,
                    "objectPosition": undefined,
                    "padding": 0,
                    "paddingRight": "3px",
                    "position": "absolute",
                    "right": 0,
                    "top": 0,
                    "width": 0,
                  }
                }
              />
            </noscript>
          </span>
          Countinue with Twitter
        </button>
      </div>
    </div>
  </div>
</div>
`;
